scalyr_agent/configuration.py

```diff
@@ -3329,12 +3329,26 @@ def __verify_logs_and_monitors_configs_and_apply_defaults(self, config, file_pat
 
         i = 0
         for log_entry in config.get_json_array("journald_logs"):
-            self.__verify_log_entry_with_key_and_set_defaults(
-                log_entry,
-                key="journald_unit",
-                config_file_path=file_path,
-                entry_index=i,
-            )
+            bad_config = False
+            try:
+                self.__verify_log_entry_with_key_and_set_defaults(
+                    log_entry,
+                    key="journald_unit",
+                    config_file_path=file_path,
+                    entry_index=i,
+                )
+            except BadConfiguration:
+                bad_config = True
+
```

Comment:
So it looks like we fall back to trying to find ``journald_globs`` entry if ``journald_unit`` fails validation (and in such case we would also use default value for ``journald_unit`` in the monitor, right?).

I think that's probably fine in most scenarios, but I guess we could also be masking some of the potential issues (e.g. unintentional typo / wrong field value for ``journald_unit`` config option).

So I wonder how we could handle that better.

Perhaps store ``BadConfiguration` error for ``journald_unit`` and re-throw that in case ``journald_globs`` code path returns empty / default value?

I guess that would be better than nothing, but we could still be masking one error scenario (when user tries to define both options, but the first one contains a typo / similar - or would that be handled correctly later on in the monitor code?).

Reply:
I'll have a think to see if there's a better way to solve this.

Reply:
For now I made a small change to just always log a message in case we end up here.

It's not the most ideal thing, but it's probably better than nothing. Probably an improvement would be to just log a messages in case the validation fails and the value is also empty.